star-rating/src/components/StarRatingField.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+import { useFieldPlugin } from '../useFieldPlugin'
+import StarIcon from './StarIcon.vue'
+
+const plugin = useFieldPlugin()
+
+const model = ref(0);
+const hoverValue = ref(0)
+const hoverActive = ref(false)
+const numberOfStars = ref(5)
+
+if (plugin.data.options.numberOfStars) {
```

Comment:
maybe here we also need to check if the  `plugin.data.options.numberOfStars` is a number otherwise parseInt might throw an error